common/lib/xmodule/xmodule/tests/test_word_cloud.py

```diff
@@ -38,15 +38,15 @@ def test_good_ajax_request(self):
             response['student_words'],
             {'sun': 1, 'dog': 6, 'cat': 12}
         )
-        self.assertListEqual(
-            response['top_words'],
-            [{'text': 'dad', 'size': 2, 'percent': 9.0},
-             {'text': 'sun', 'size': 1, 'percent': 5.0},
-             {'text': 'dog', 'size': 6, 'percent': 27.0},
-             {'text': 'mom', 'size': 1, 'percent': 5.0},
-             {'text': 'cat', 'size': 12, 'percent': 54.0}]
-        )
 
-        self.assertEqual(
-            100.0,
-            sum(i['percent'] for i in response['top_words']))
+        expected_top_words = [{'text': 'dad', 'size': 2, 'percent': 9},
+                              {'text': 'sun', 'size': 1, 'percent': 4},
+                              {'text': 'dog', 'size': 6, 'percent': 27},
+                              {'text': 'mom', 'size': 1, 'percent': 5},
+                              {'text': 'cat', 'size': 12, 'percent': 55}]
+
+        for item in response['top_words']:
+            self.assertDictEqual(item, next(
+                expected_item for expected_item in expected_top_words if expected_item["text"] == item["text"]))
```

Comment:
This is still a very complex way of getting at the expected top words, I think a better thing to do is to just update the above list of `expected_top_words` to be a dictionary that maps the value of text to the exact contents of the dictionary.  Then here you can just do: `self.assertDictEqual(item, expected_top_words[item['text']])`

Does that make sense?

Reply:
@feanil I've updated the comparison back to how it was working before, the problem was with the order of response items was different between python 2 and 3, now the response items are ordered alphabetically and work with both python versions.